lib/puppet/provider/package/gem.rb

```diff
@@ -15,7 +15,17 @@
 
   has_feature :versionable, :install_options, :uninstall_options
 
-  commands :gemcmd => "gem"
+  if Puppet.features.microsoft_windows?
+    # Puppet on Windows prepends its paths to PATH. One of those paths includes Puppet's vendored ruby.
+    # That results in the which() method in self.command(name) in lib/puppet/provider.rb always returning Puppet's gem command.
+    # To prevent this, call which("gem") with a PATH without Puppet's paths, and set gemcmd to the full-path result. 
+    path_array = Puppet::Util.get_env("PATH").split(File::PATH_SEPARATOR)
+    path_array.reject! {|p| p =~ /puppet/i}
```

Comment:
The method of constructing the PATH is a bit imprecise - a few problems I see:

* What if a user relocated the installation to something without `puppet` in it?
* What if the Ruby installation path contains `puppet` in it? (admittedly less likely)

We can add a little bit of code to find the actual current running Ruby like this:
https://stackoverflow.com/a/25929455

I can help out with putting that in the right spot within Puppet.